### PR TITLE
fixing issue 9583 - issue in cascadeclassifier code clipObjects

### LINKED ISSUE #9583: Opencv Clean Build raising assertion error in cascade detector
##### System information (version)
- OpenCV => 3.3
- Operating System / Platform => Windows 10 64 Bit
- Compiler => Visual Studio 2017

Clean Build of Opencv raises assertion error in sample cascade detector C++ program.  I have checked my build twice, still have not found a solution. 

##### Detailed description
I built opencv binaries from scratch and it raises this assertion error in line
```face_cascade.detectMultiScale( frame_gray, faces, 1.1, 2, 0|CASCADE_SCALE_IMAGE, Size(30, 30) );```

I have not changed the sample program. I made sure I have the required xml files.

### PR DESCRIPTION
**Pull request by StevenPuttemans**
resolves #9583

### CODE REVIEW

File: modules/objdetect/src/cascadedetect.cpp [hunk 1]
```diff
@@ -1666,7 +1666,7 @@ void clipObjects(Size sz, std::vector<Rect>& objects,
     for( i = 0; i < n; i++ )
     {
         Rect r = win0 & objects[i];
-        if( r.area() > 0 )
+        if( r.empty() > 0 )
```
Comment (alalek) [inline modules/objdetect/src/cascadedetect.cpp]: Condition should be `!r.empty()` (we need non-empty rectangles)

Comment (StevenPuttemans) [reply modules/objdetect/src/cascadedetect.cpp]: I will fix it this afternoon... Simpky a bad copy paste 😀

**Comment by StevenPuttemans** [comment]
@alalek completely forgot about this yesterday but should be fixed now!

File: modules/objdetect/src/cascadedetect.cpp [hunk 2]
```diff
@@ -1666,7 +1666,7 @@ void clipObjects(Size sz, std::vector<Rect>& objects,
     for( i = 0; i < n; i++ )
     {
         Rect r = win0 & objects[i];
-        if( r.area() > 0 )
+        if( !r.empty() > 0 )
```
Comment (alalek) [inline modules/objdetect/src/cascadedetect.cpp]: ` > 0` is not needed here

Comment (StevenPuttemans) [reply modules/objdetect/src/cascadedetect.cpp]: Oops fix on it's way!

**Comment by vpisarev** [comment]
:+1:

**Pull request merged**